### DISCUSSION TITLE
how to run nicegui with gunicorn not running

### DISCUSSION THREAD

**Discussion by bobwatcherx** [zauberzeug/nicegui#971] [Q&A]
### Question

I want to deploy with gunicorn, is there a correct way to write it to run with uvicorn or gunicorn
```py
from flask import Flask
from nicegui import ui

app = Flask(__name__)

@app.route("/")
def index():
    app = with ui.column():
        ui.label("test").style("font-size:50px;font-weight:bold")
        with ui.row().classes("bg-red-300 w-full p-10 justify-between"):
            ui.button("sample test"),
            ui.label(0),
            ui.button("Sa")
    return app
if __name__ == "__main__":
    app.run()

```

**Comment by bobwatcherx** [comment]
add more documentation about it

**Comment by bobwatcherx** [comment]
I want to deploy to vercel but I tested it locally but it doesn't work and I got an error like this.
my code here
 https://github.com/bobwatcherx/niceVercel.git

```
root@miop-HP-Convertible-x360-11-ab1XX:/home/miop/belajar/nicepro# vercel dev
Vercel CLI 30.0.0
> Ready! Available at http://localhost:3000
> Building @vercel/python:main.py
Installing required dependencies...
> Built @vercel/python:main.py [2s]
Error: spawn /root/.cache/com.vercel.fun/runtimes/python3/bootstrap EACCES
    at Process.ChildProcess._handle.onexit (node:internal/child_process:285:19)
    at onErrorNT (node:internal/child_process:483:16)
    at processTicksAndRejections (node:internal/process/task_queues:82:21)
Error: An unexpected error occurred!
Error: spawn /root/.cache/com.vercel.fun/runtimes/python3/bootstrap EACCES
    at Process.ChildProcess._handle.onexit (node:internal/child_process:285:19)
    at onErrorNT (node:internal/child_process:483:16)
    at processTicksAndRejections (node:internal/process/task_queues:82:21)

```

**Comment by falkoschindler** [comment]
My thoughts on this:

1. I'm not sure whether you want to run _with_ or _without_ gunicorn/uvicorn. Title and description seem to contradict each other:
    > how to run nicegui with gunicorn not running
    > is there a correct way to write it to run with uvicorn or gunicorn
2. `app = with ui.column():` looks like a syntax error to me.
3. Although NiceGUI [integrates with existing FastAPI apps](https://github.com/zauberzeug/nicegui/tree/main/examples/fastapi), to my knowledge there is currently no way to combine it with Flask.
